Sources/KlaviyoUI/KlaviyoWebView/KlaviyoWebWrapperViewController.swift

```diff
@@ -0,0 +1,169 @@
+//
+//  Created by Andrew Balmer on 10/22/24.
+//
+
+import Foundation
+import UIKit
+
+public class KlaviyoWebWrapperViewController: UIViewController {
+    // MARK: - Properties
+
+    let viewModel: KlaviyoWebViewModeling
+    let style: KlaviyoWebWrapperStyle
+
+    private lazy var dismissGestureRecognizer: UITapGestureRecognizer = {
+        let tapRecognizer = UITapGestureRecognizer(target: self, action: #selector(handleDismissGesture))
+        tapRecognizer.numberOfTapsRequired = 1
+        tapRecognizer.numberOfTouchesRequired = 1
+        return tapRecognizer
+    }()
+
+    // MARK: - Subviews
+
+    private lazy var blurEffectView: UIVisualEffectView? = {
+        guard case let .blurred(effect) = style.backgroundStyle else { return nil }
+
+        let blurEffect = UIBlurEffect(style: effect)
+        let blurEffectView = UIVisualEffectView(effect: blurEffect)
+
+        blurEffectView.addGestureRecognizer(dismissGestureRecognizer)
+
+        return blurEffectView
+    }()
+
+    private lazy var tintView: UIView? = {
+        guard case let .tinted(color, opacity) = style.backgroundStyle else { return nil }
+
+        let tintView = UIView()
+        tintView.backgroundColor = color
+        tintView.layer.opacity = opacity
+
+        tintView.addGestureRecognizer(dismissGestureRecognizer)
+
+        return tintView
+    }()
+
+    private lazy var shadowContainerView: UIView? = {
+        guard let shadowProperties = style.shadowStyle else { return nil }
+
+        let shadowContainerView = UIView()
+
+        shadowContainerView.layer.shadowColor = shadowProperties.color
+        shadowContainerView.layer.shadowOpacity = shadowProperties.opacity
+        shadowContainerView.layer.shadowOffset = shadowProperties.offset
+        shadowContainerView.layer.shadowRadius = shadowProperties.radius
+
+        shadowContainerView.layer.masksToBounds = false
+
+        return shadowContainerView
+    }()
+
+    private lazy var webViewController: KlaviyoWebViewController = {
+        let webViewController = KlaviyoWebViewController(viewModel: viewModel)
+        guard let webView = webViewController.view else { return webViewController }
+        webView.layer.cornerRadius = 16
+        webView.layer.masksToBounds = true
+
+        return webViewController
+    }()
+
+    // MARK: - View Initialization
+
+    init(viewModel: KlaviyoWebViewModeling, style: KlaviyoWebWrapperStyle = .default) {
+        self.viewModel = viewModel
+        self.style = style
+
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    @available(*, unavailable)
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override public func viewDidLoad() {
+        super.viewDidLoad()
+        loadSubviews()
+    }
+
+    func loadSubviews() {
+        if let blurEffectView {
+            view.addSubview(blurEffectView)
+            blurEffectView.pin(to: view)
+        }
+
+        if let tintView {
```

Comment:
Is this the new way to check for nil?

Reply:
yeah, no need to do `if let tintView = tintView` anymore